cgsmiles/graph_layout.py

```diff
@@ -0,0 +1,222 @@
+"""
+Generate positions for 2D atomistic graphs.
+"""
+from collections import defaultdict, OrderedDict
+import math
+import numpy as np
+import scipy.optimize
+from scipy.spatial.distance import pdist, squareform
+import networkx as nx
+
+def find_triplets(graph):
+    """
+    Find all triplets of nodes in the graph.
+
+    Parameters
+    ----------
+    graph: nx.Graph
+
+    Returns
+    -------
+    tuple([abc.hashable,abc.hashable,abc.hashable])
+        the node keys corresponding to the triplet
+        the central node is the common one.
+    """
+    triplets = set()
+    for edge1 in graph.edges:
+        for edge2 in graph.edges:
+            if edge1 != edge2:
+                common_node = set(edge1).intersection(edge2)
+                if common_node:
+                    common_node = common_node.pop()  # Extract the common node
+                    remaining_nodes = list(set(edge1 + edge2) - {common_node})
+                    triplet = (remaining_nodes[0], common_node, remaining_nodes[1])
+                    triplets.add(triplet)
+    return triplets
```

Comment:
E**2 is kind of painful. Probably /way/ more efficient to iterate over nodes with `degree>1` and doing an `itertools.combinations` on their neighbours

Reply:
let's leave this as MAINT